### PR TITLE
Temporary stack variable being stored as the wrong type

### PR DESCRIPTION
**Pull request by JohnDumbell**
- [x] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [x] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

In a very particular instance (shown in the unit test) the type of the temporary stack variable got changed from what it was originally to the type of whatever expression triggered the save_stack_entries call. This just makes sure the type of the stored variable doesn't change.

### CODE REVIEW

File: jbmc/regression/jbmc/temp_stack_variable_packing/test.desc [hunk 1]
```diff
@@ -0,0 +1,6 @@
+CORE
+Primary.class
+--function stuff.array_issue.Primary.Run
+^VERIFICATION SUCCESSFUL|VERIFICATION FAILED$
```
Comment (smowton) [inline jbmc/regression/jbmc/temp_stack_variable_packing/test.desc]: We really don't know which one to expect?

File: jbmc/src/java_bytecode/java_bytecode_convert_method.cpp [hunk 2]
```diff
@@ -3241,7 +3241,8 @@ void java_bytecode_convert_methodt::save_stack_entries(
     }
     if(replace)
     {
-      create_stack_tmp_var(tmp_var_prefix, tmp_var_type, block, stack_entry);
+      create_stack_tmp_var(
```
Comment (smowton) [inline jbmc/src/java_bytecode/java_bytecode_convert_method.cpp:3244]: No functional changes?

Comment (JohnDumbell) [reply jbmc/src/java_bytecode/java_bytecode_convert_method.cpp:3244]: Style commit I just pushed got rid of the change for some reason - fixed now.

File: jbmc/regression/jbmc/temp_stack_variable_packing/test.desc (hunk 1 above)
Comment (JohnDumbell) [reply jbmc/regression/jbmc/temp_stack_variable_packing/test.desc]: We don't really mind, as long as it completes in some form the fix will still be in place (it just exceptioned half way through before).

Comment (smowton) [reply jbmc/regression/jbmc/temp_stack_variable_packing/test.desc]: True, but generally we should know what our tests are doing and specify that. In this case I recommend putting an `assert(false)` in the `Accept` method and checking that we have `VERIFICATION FAILED` and also that we failed that particular assertion, thus establishing that the test explored the path where we make the call (cf. failing an earlier null-pointer check or something) as we intended.

**Review by smowton — APPROVED** [review]
Fine with me once the test is firmed up as noted above

**Review by allredj — COMMENTED** [review]
This PR failed Diffblue compatibility checks (cbmc commit: 0bb165d1bbb99021d8b0cc3981f8415501df4d8d).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/88978662
Status will be re-evaluated on next push.
Please contact @peterschrammel, @thk123, or @allredj for support.

Common spurious failures:
  - the cbmc commit has disappeared in the mean time (e.g. in a force-push)
  - the author is not in the list of contributors (e.g. first-time contributors).

The incompatibility may have been introduced by an earlier PR. In that case merging this
PR should be avoided unless it fixes the current incompatibility.

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: b9392c548b6ac26d8787d6bb0406dbe97190b18c).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/88986777

**Review by peterschrammel — APPROVED** [review]

**Pull request merged**